app/src/main/java/com/goayo/debtify/model/database/MockDatabase.java

```diff
@@ -0,0 +1,108 @@
+package com.goayo.debtify.model.database;
+
+import com.goayo.debtify.model.Group;
+import com.goayo.debtify.model.User;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ *
+ * @Author Oscar Sanner
+ * @date 2020-09-15
+ *
+ * <p>
+ * A mock class to try different database calls on, without connecting to an actual database.
+ */
+
+class MockDatabase implements IDatabase {
+
+    private Set<Group> groups;
+    private Set<User> users;
+
+    public MockDatabase(){
+        users = new HashSet<>();
+        users.add(new User("0704345621", "Olle Johansson"));
+        users.add(new User("0735216752", "Rickard Nicklasson"));
+        users.add(new User("0734355982", "Gabriel Phu"));
+        users.add(new User("0773345654", "Yenan Sjögren"));
+
+        groups = new HashSet<>();
+        groups.add(new Group("Afterwork", "1002", users));
+        groups.add(new Group("Trip To Spain", "1003", users));
+        groups.add(new Group("School friends", "1004", users));
+    }
+
+
+    /**
+     * A method for retrieving groups from the database.
+     *
+     * @param phoneNumber The phone number for the user for who's groups will be returned.
+     *
+     * @return All groups associated with the phone number sent into the method.
+     */
+    @Override
+    public Set<Group> getGroups(String phoneNumber) {
+        return groups;
+    }
+
+    /**
+     * A method which retries a User from the database, if the sent in phone number is associated with a user.
+     *
+     * @param phoneNumber The phone number for the user which the database will look for.
+     *
+     * @return The user in the database who has the phone number
+     */
+    @Override
+    public User getUser(String phoneNumber) {
+        for (User u : users){
+            if(u.getPhoneNumber().equals(phoneNumber)){
+                return u;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * A method registering a user to the database
+     *
+     * @param phoneNumber Phone number for the user to be registered in the database.
+     * @param password Password for the user to be registered in the database.
+     * @param name Name for the user to be registered in the database.
+     *
+     * @return true if the registration was successful.
+     */
+    @Override
+    public boolean registerUser(String phoneNumber, String password, String name) {
+        users.add(new User(phoneNumber, name));
+        return true;
+    }
```

Comment:
Should maybe return false if user has already been created. Should we give this responsibility to database instead of account?

Reply:
This is a really good question, what would make the most sense in terms of conceptual responsibilities?

Reply:
Both programming pairs agree that ModelEngine having database makes more sense conceptually. Will make a change-request on the ModelEngine pullrequest to add a database property there. Will thus remove the database property from Account class.